GiniVision/Classes/Core/GiniScreenAPICoordinator+Camera.swift

```diff
@@ -0,0 +1,292 @@
+            return true
+        } else if giniConfiguration.onboardingShowAtLaunch {
+            return true
+        }
+        
+        return false
+    }
+    
+    private func showOnboardingScreen() {
+        cameraViewController?.hideCameraOverlay()
+        cameraViewController?.hideCaptureButton()
+        cameraViewController?.hideFileImportTip()
+        
+        let vc = OnboardingContainerViewController { [weak self] in
+            guard let `self` = self else { return }
+            self.cameraViewController?.showCameraOverlay()
+            self.cameraViewController?.showCaptureButton()
+            self.cameraViewController?.showFileImportTip()
+        }
+        
+        let navigationController = UINavigationController(rootViewController: vc)
+        navigationController.applyStyle(withConfiguration: giniConfiguration)
+        navigationController.modalPresentationStyle = .overCurrentContext
+        screenAPINavigationController.present(navigationController, animated: true, completion: nil)
+    }
+    
+    private func showMultipageNewFeatureDialog() {
+        let alertDialog = AlertDialogController(giniConfiguration: giniConfiguration,
+                                                title: "This is the title",
+                                                message: "This is the subtitle",
+                                                image: UIImageNamedPreferred(named: "multipageIcon"),
+                                                buttonTitle: "Let's scan!",
+                                                buttonImage: UIImage(named: "cameraIcon",
+                                                                     in: Bundle(for: GiniVision.self),
+                                                                     compatibleWith: nil))
+        alertDialog.continueAction = {
+            alertDialog.dismiss(animated: true, completion: nil)
+            AlertDialogController.shouldShowNewMultipageFeature = false
+        }
+        alertDialog.cancelAction = alertDialog.continueAction
+        screenAPINavigationController.present(alertDialog,
+                                              animated: true,
+                                              completion: nil)
+    }
+    
+    func showNextScreenAfterPicking(documents: [GiniVisionDocument]) {
+        if let firstDocument = documents.first, let documentsType = documents.type {
+            switch documentsType {
+            case .image:
+                if let imageDocuments = self.visionDocuments as? [GiniImageDocument],
+                    let lastDocument = imageDocuments.last {
+                    if self.giniConfiguration.multipageEnabled {
+                        if let imageDocuments = self.visionDocuments as? [GiniImageDocument],
+                            lastDocument.isImported {
+                            self.showMultipageReview(withImageDocuments: imageDocuments)
```

Comment:
A lot of if cases here. Maybe it could have been expressed more clearly? Maybe a combination of comments and splitting the logic in several methods?

Reply:
In another branch I modified this if cases since it changes a little bit. Also, if you pay more attention to this, there is a redundant conditional unwrapping for the `imageDocuments`.
I won't change nothing here because it is already modify in another branch.